### PR TITLE
Feat block countries from certain regions

### PR DESCRIPTION
**Pull request by vermakhushboo**
## What does this PR do?

Add logic in `auth` group to block requests from certain IP Addresses

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/shared/api.php [hunk 1]
```diff
@@ -592,3 +593,19 @@
...
     });
+
+App::init()
+    ->groups(['restrict'])
+    ->inject('request')
+    ->inject('geodb')
+    ->action(function (Request $request, Reader $geodb) {
+        if (!empty(app::getEnv('_APP_RESTRICTED_COUNTRIES', ''))) {
+            $countries = explode(',', App::getEnv('_APP_RESTRICTED_COUNTRIES', ''));
+            $record = $geodb->get($request->getHeader('x-forwarded-for'));
```
Comment (Meldiron) [inline app/controllers/shared/api.php:604]:
```suggestion
            $record = $geodb->get($request->getIP());
```

**Review by eldadfux — COMMENTED** [review]
Not sure we need to impose the limit on the users API. Also, I think we can settle on limiting access to the console and leave end-users signup open. 

Since this will be a console only limit, I would also consider using the existing pattern of `_APP_CONSOLE_WHITELIST` for the env prefix.

File: .env [hunk 2]
```diff
@@ -103,3 +103,4 @@ _APP_ASSISTANT_OPENAI_API_KEY=
 _APP_MESSAGE_SMS_TEST_DSN=
 _APP_MESSAGE_EMAIL_TEST_DSN=
 _APP_MESSAGE_PUSH_TEST_DSN=
+_APP_RESTRICTED_COUNTRIES=AQ
```
Comment (christyjacob4) [inline .env:106]:
```suggestion
_APP_COUNTRIES_DENYLIST=AQ
```

File: app/controllers/shared/api.php [hunk 3]
```diff
@@ -592,3 +593,19 @@
             throw new Exception(Exception::GENERAL_USAGE_DISABLED);
         }
     });
+
+App::init()
+    ->groups(['restrict'])
+    ->inject('request')
+    ->inject('geodb')
+    ->action(function (Request $request, Reader $geodb) {
+        if (!empty(app::getEnv('_APP_RESTRICTED_COUNTRIES', ''))) {
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:602]:
```suggestion
        $denylist = App::getEnv('_APP_RESTRICTED_COUNTRIES', '');
        if (!empty($denylist)) {
```

File: app/controllers/shared/api.php [hunk 4]
```diff
@@ -592,3 +593,19 @@
...
+            $countries = explode(',', App::getEnv('_APP_RESTRICTED_COUNTRIES', ''));
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:603]:
```suggestion
            $countries = explode(',', $denylist);
```

File: app/controllers/shared/api.php [hunk 5]
```diff
@@ -592,3 +593,19 @@
...
+            $record = $geodb->get($request->getIP());
+            $country = $record['country']['iso_code'];
+            $countryName = $record['country']['names']['en'];
+            if (in_array($country, $countries)) {
+                throw new Exception(Exception::GENERAL_ACCESS_FORBIDDEN, "Sorry, access from $countryName is restricted");
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:608]: We need to return a 503 service unavailable error in this case. Check the errors.php file to see if we have something like that already 👍

File: tests/e2e/Services/Account/AccountBase.php [hunk 6]
```diff
@@ -43,6 +43,20 @@ public function testCreateAccount(): array
         /**
          * Test for FAILURE
          */
+        $response = $this->client->call(Client::METHOD_POST, '/account', array_merge([
```
Comment (christyjacob4) [inline tests/e2e/Services/Account/AccountBase.php:46]: Lets add a comment indicating what we are doing in this test case

File: app/controllers/shared/api.php (hunk 5 above)
Comment (vermakhushboo) [reply app/controllers/shared/api.php:608]: We don't have any relevant 503 errors at the moment. We'd need a new one.

Comment (vermakhushboo) [reply app/controllers/shared/api.php]: Creating one with 451 as discussed

**Review by Meldiron — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

**Pull request merged**